src/containers/App/App.test.js

```diff
@@ -39,16 +41,69 @@ describe('App', () => {
   })
 
   it('should match the correct snapshot if there is an error', () => {
-    let mockProps2 = {
-      notes: [{ id: 1, title: 'Note Title' }],
-      items: [],
-      error: 'Something went wrong',
-      fetchNotes: jest.fn()
-    }
-    wrapper = shallow(<App {...mockProps2} />)
+    wrapper.setProps({error: 'Something went wrong'})
+    wrapper = shallow(<App {...mockProps} />)
     expect(wrapper).toMatchSnapshot()
   })
 
+  describe('Routes', () => {
+
+    it('should render the NoteArea container when at the root route', () => {
+      const wrapper = mount(
+        <Provider store={store}>
+          <MemoryRouter initialEntries={['/']}>
+            <App />
+          </MemoryRouter>
+        </Provider>
+      )
+      expect(wrapper.find(NoteArea)).toHaveLength(1)
+    })
+
+    it('should render the NoteForm container when at /new-note', () => {
+      const wrapper = mount(
+        <Provider store={store}>
+          <MemoryRouter initialEntries={['/new-note']}>
+            <App />
+          </MemoryRouter>
+        </Provider>
+      )
+      expect(wrapper.find(NoteForm)).toHaveLength(1)
+    })
+
+    it('should render the NoteForm container when a valid note is searched', () => {
+
+      const mockNotes = [{ id: 1, title: "Title One" }, { id: 2, title: "Title Two" }]
+      const mockItems = [{ description: 'Note one of Title one', noteID: 1, id: 1 }, { description: 'Note two of Title one', noteID: 1, id: 2 }, { description: 'Note three of Title one', noteID: 1, id: 3 },
+      { description: 'Note one of Title two', noteID: 2, id: 1 }, { description: 'Note two of Title two', noteID: 2, id: 2 }]
+      const fetchNotes = jest.fn()
+      wrapper = mount(
+        <Provider store={store}>
+          <MemoryRouter initialEntries={['/notes/1']}>
+            <App notes={mockNotes} items={mockItems} error={''} fetchNotes={fetchNotes} />
+          </MemoryRouter>
+        </Provider>
+      )
+      expect(wrapper.find(NoteForm)).toHaveLength(1)
+    })
+
+    it("should render the NotFound component when a note doesn't exist", () => {
+
+      const mockNotes = [{ id: 1, title: "Title One" }, { id: 2, title: "Title Two" }]
+      const mockItems = [{ description: 'Note one of Title one', noteID: 1, id: 1 }, { description: 'Note two of Title one', noteID: 1, id: 2 }, { description: 'Note three of Title one', noteID: 1, id: 3 },
+      { description: 'Note one of Title two', noteID: 2, id: 1 }, { description: 'Note two of Title two', noteID: 2, id: 2 }]
+      const fetchNotes = jest.fn()
+      wrapper = mount(
+        <Provider store={store}>
+          <MemoryRouter initialEntries={['/notes/3']}>
+            <App notes={mockNotes} items={mockItems} error={''} fetchNotes={fetchNotes} />
+          </MemoryRouter>
+        </Provider>
+      )
+      expect(wrapper.find(NotFound)).toHaveLength(1)
+    })
+
+  })
+
```

Comment:
Really interesting how you tested the routes here - I've never seen the MemoryRouter used before but after doing some research it definitely makes sense here!